File: Chapter9ActivityHook2/app/src/main/java/com/mars/activityhook2/HostApplication.java
package com.mars.activityhook2;

import android.app.Application;
import android.content.Context;

/**
 * Created by geyan on 2020/10/31
 */
public class HostApplication extends Application {

    private static Context mContext;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        mContext = base;
    }

    public static Context getContext() {
        return mContext;
    }
}

File: Chapter9ZeusStudy1_2/app/src/main/java/com/mars/zeusstudy1_2/HostApplication.kt
package com.mars.zeusstudy1_2

import android.app.Application
import android.content.Context
import com.mars.pluginservice.PluginManager

/**
 * Created by geyan on 2020/11/21
 */
class HostApplication : Application() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        PluginManager.init(this)

        AMSHookHelper.hookAMN()
        AMSHookHelper.hookActivityThread()
    }
}